drivers/timer/nrf_rtc_timer.c

```diff
@@ -160,10 +160,10 @@ static void set_absolute_alarm(uint32_t chan, uint32_t abs_val)
 	uint32_t cc_val = abs_val & COUNTER_MAX;
 	uint32_t prev_cc = get_comparator(chan);
 
-
 	do {
 		now = counter();
 
+		event_enable(chan);
```

Comment:
This seems OK.  My concern was that this could be executed with interrupts enabled.  It can't from `z_clock_set_timeout()` because that holds a lock.  It can't for other channels because `z_nrf_rtc_timer_compare_set()` also holds a lock.  

However, while this will be fine, it seems that anybody who uses `z_nrf_rtc_timer_compare_int_lock()` then invokes `z_nrf_rtc_timer_compare_set()` will have their lock cleared prematurely when the set completes.  That seems like a bug which I'll raise elsewhere.

Reply:
No, I see the `key` parameter should avoid that.